File: csc454hw06/src/main/java/Pipe.java
public class Pipe <T> {

    Port<T> sending;
    Port<T> receiving;

    public Pipe(Port<T> outgoingPort, Port<T>receivingPort) {
        this.sending = outgoingPort;
        this.receiving = receivingPort;
    }

    /**
     * Move the value from the sendingPort to the receiving port.
     */
    public void passValue(){
        receiving.currentValue = sending.currentValue;
    }

}

File: csc454hw06_cpp/Pipe.cpp
#ifndef PIPE
#define PIPE

template <class T>
class Pipe
{
public:
    Port<T> *sending;
    Port<T> *receiving;

    Pipe(Port<T> *outgoingPort, Port<T> *receivingPort)
    {
        sending = outgoingPort;
        receiving = receivingPort;
    }

    /**
     * Move the value from the sendingPort to the receiving port.
     */
    void passValue()
    {
        receiving->currentValue = sending->currentValue;
    }
};

#endif